google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableDefinition.java

```diff
@@ -204,7 +206,7 @@ final boolean baseEquals(TableDefinition tableDefinition) {
 
   Table toPb() {
     Table tablePb = new Table();
-    if (schema != null) {
+    if (schema.getFields().size() != 0) {
```

Comment:
Without this change, we'll try to set schema of a view table we want to create, which is not allowed.
This concerns me a bit. Previously we have null to say "we don't know what the schema is" and drop it from the request. 
Perhaps we should let the field be nullable and do a null check on `getSchema`? I'm not sure what the right answer is.

Reply:
This check before populating is similar to how it is done in Python, not that that makes it right.